From f2bc1367496c0c168fa75df9204a0c0886a4981a Mon Sep 17 00:00:00 2001
From: Sticky Chen <sticky.chen@mediatek.com>
Date: Mon, 11 Feb 2019 14:31:22 +0800
Subject: [PATCH] Prevent scan searching overhead

If user connects to a AP but supplicant fails to
connect it successfully, supplicant will trigger
periodical scan to search the AP. The searching
scan brings redudant power consumption under
screen off.
To prevent the overhead, we remove networks to
supplicant before starting PNO scan.

Test: Connect to a nonexistent AP, screen off and
make sure there is no searching scan triggered by
supplicant.
Bug: 124281812
Change-Id: I4a0e99ea1df4fee0b0e3157ed0b82bc83d379765
---
 .../com/android/server/wifi/scanner/WificondScannerImpl.java     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/service/java/com/android/server/wifi/scanner/WificondScannerImpl.java b/service/java/com/android/server/wifi/scanner/WificondScannerImpl.java
index 41ff9a5ab..68dce534e 100644
--- a/service/java/com/android/server/wifi/scanner/WificondScannerImpl.java
+++ b/service/java/com/android/server/wifi/scanner/WificondScannerImpl.java
@@ -411,6 +411,7 @@ public class WificondScannerImpl extends WifiScannerImpl implements Handler.Call
     }
 
     private boolean startHwPnoScan(WifiNative.PnoSettings pnoSettings) {
+        mWifiNative.removeAllNetworks(mIfaceName);
         return mWifiNative.startPnoScan(mIfaceName, pnoSettings);
     }
 
-- 
2.17.1

